### PR TITLE
Remove readonly on auto-property when a setter is needed

### LINKED ISSUE #47999: IDE0032 fixer should strip readonly keyword from properties
The code fix provider for IDE0032 (use auto-property) produces invalid code when faced with C# 8 readonly instance members.

**Version Used**: 3.7.0-6.20459.4 (7ee7c540), Visual Studio 16.7.4

**Steps to Reproduce**:

1. Minimal reproduction:

```csharp
struct IDE0032
{
    private int x;
    public readonly int X => x;
    public void SetX(int value) => x = value; // ensures the produced auto-property is settable
}
```

2. IDE0032 will be reported on `x`. Apply the "Use auto property" code fix.

**Expected Behavior**:

```csharp
struct IDE0032
{
    public int X { get; private set; }
    public void SetX(int value) => X = value;
}
```

**Actual Behavior**:

```csharp
struct IDE0032
{
    public readonly int X { get; private set; } // CS8659: Auto-implemented property 'IDE0032.X' cannot be marked 'readonly' because it has a 'set' accessor.
    public void SetX(int value) => X = value;
}
```

### PR DESCRIPTION
**Pull request by louis-z**
Fixes #47999

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs [hunk 1]
```diff
@@ -2447,6 +2447,40 @@ int P
...
+    [|int i|];
+    public readonly int P => i;
+    public void SetP(int value) => i = value;
+}",
+@"class Class
+{
+    public int P { get; private set; }
+    public void SetP(int value) => P = value;
+}");
+        }
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs:2469]: can you add a test without hte accessibility modifier?  i.e. `readonly int P => 1;`  I want to make sure removing that modifier doesn't lead to very weirdly formatted code.  thanks!

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
LMK when this is green, and i'll merge in.  Thanks! :)

File: src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs [hunk 2]
```diff
@@ -2447,6 +2447,58 @@ int P
...
+        {
+            await TestInRegularAndScript1Async(
+@"class Class
+{
+    [|int i|];
+    public readonly int P => i;
+}",
+@"class Class
+{
+    public readonly int P { get; }
```
Comment (davidwengier) [inline src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs:2501]: Am I missing something? This isn't valid C#. Shouldn't all of these test cases be structs?

Comment (louis-z) [reply src/Analyzers/CSharp/Tests/UseAutoProperty/UseAutoPropertyTests.cs:2501]: Oops, my bad. I'll change that. Good catch! 😉

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**